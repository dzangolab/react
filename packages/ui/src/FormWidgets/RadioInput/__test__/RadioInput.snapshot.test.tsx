import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { RadioInput } from "..";

const options = [
  { label: "One", value: "1" },
  { label: "Two", value: "2" },
  { label: "Three", value: "3" },
];

describe("RadioInput", () => {
  test("should match snapshot with options", () => {
    const { container } = render(<RadioInput options={options} />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with disabled set to true", () => {
    const { container } = render(
      <RadioInput options={options} disabled={true} />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with custom className", () => {
    const { container } = render(
      <RadioInput options={options} className="custom-class" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with a custom label", () => {
    const { container } = render(
      <RadioInput options={options} label="Custom label" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with a custom name", () => {
    const { container } = render(
      <RadioInput options={options} name="radioInput" />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with all props set", () => {
    const { container } = render(
      <RadioInput
        className="radio-input"
        name="radioInput"
        label="Custom Label"
        options={options}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
