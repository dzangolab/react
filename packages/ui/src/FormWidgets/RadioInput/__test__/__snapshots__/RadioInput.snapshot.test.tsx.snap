// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioInput > should match snapshot with a custom label 1`] = `
<div>
  <fieldset
    class="field"
  >
    <legend>
      Custom label
    </legend>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="1"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="2"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="3"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`RadioInput > should match snapshot with a custom name 1`] = `
<div>
  <fieldset
    class="field"
  >
    <legend />
    <div
      class="radio-button-wrapper"
    >
      <input
        id="1"
        name="radioInput"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="2"
        name="radioInput"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="3"
        name="radioInput"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`RadioInput > should match snapshot with all props set 1`] = `
<div>
  <fieldset
    class="field radio-input"
  >
    <legend>
      Custom Label
    </legend>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="1"
        name="radioInput"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="2"
        name="radioInput"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="3"
        name="radioInput"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`RadioInput > should match snapshot with custom className 1`] = `
<div>
  <fieldset
    class="field custom-class"
  >
    <legend />
    <div
      class="radio-button-wrapper"
    >
      <input
        id="1"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="2"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="3"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`RadioInput > should match snapshot with disabled set to true 1`] = `
<div>
  <fieldset
    class="field"
  >
    <legend />
    <div
      class="radio-button-wrapper"
    >
      <input
        disabled=""
        id="1"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        disabled=""
        id="2"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        disabled=""
        id="3"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`RadioInput > should match snapshot with options 1`] = `
<div>
  <fieldset
    class="field"
  >
    <legend />
    <div
      class="radio-button-wrapper"
    >
      <input
        id="1"
        type="radio"
        value="1"
      />
      <label
        for="1"
      >
        One
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="2"
        type="radio"
        value="2"
      />
      <label
        for="2"
      >
        Two
      </label>
    </div>
    <div
      class="radio-button-wrapper"
    >
      <input
        id="3"
        type="radio"
        value="3"
      />
      <label
        for="3"
      >
        Three
      </label>
    </div>
  </fieldset>
</div>
`;
